src/components/Api.js

```diff
@@ -0,0 +1,134 @@
+export default class Api {
+    constructor({ baseUrl, headers }) {
+        this._baseUrl = baseUrl;
+        this._headers = headers;
+    }
+
+    _handleResponse(res) {
+        if (res.ok) {
+          return res.json().catch((err) => {
+            console.error("Error parsing JSON:", err);
+            return Promise.reject("Failed to parse JSON response");
+          });
+        }
+        return Promise.reject(`Error: ${res.status}`);
+      }
+
+
+    getInitialCards() {
+        return fetch(`${this._baseUrl}/cards`, {
+          method: "GET",
+          headers: this._headers,
+        })
+        .then(this._handleResponse)
+        .then((cards) => {
+          console.log("Fetched initial cards:", cards);
+          return cards;
+        })
+        .catch((err) => {
+          console.error("Error fetching initial cards:", err);
```

Comment:
You should **not** use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request, and this end is in `index.js`  after all `then` blocks. And in real projects the `catch` blocks usually open error popups and so on.